rust/src/io/exec/planner.rs

```diff
@@ -173,6 +176,67 @@ impl Planner {
+        match data_type {
+            SQLDataType::String => Ok(ArrowDataType::Utf8),
+            SQLDataType::Binary(_) => Ok(ArrowDataType::Binary),
+            SQLDataType::Float(_) => Ok(ArrowDataType::Float32),
+            SQLDataType::Double => Ok(ArrowDataType::Float64),
+            SQLDataType::Boolean => Ok(ArrowDataType::Boolean),
+            SQLDataType::TinyInt(_) => Ok(ArrowDataType::Int8),
+            SQLDataType::SmallInt(_) => Ok(ArrowDataType::Int16),
+            SQLDataType::Int(_) | SQLDataType::Integer(_) => Ok(ArrowDataType::Int32),
+            SQLDataType::BigInt(_) => Ok(ArrowDataType::Int64),
+            SQLDataType::UnsignedTinyInt(_) => Ok(ArrowDataType::UInt8),
+            SQLDataType::UnsignedSmallInt(_) => Ok(ArrowDataType::UInt16),
+            SQLDataType::UnsignedInt(_) => Ok(ArrowDataType::UInt32),
+            SQLDataType::UnsignedBigInt(_) => Ok(ArrowDataType::UInt64),
+            SQLDataType::Date => Ok(ArrowDataType::Date32),
+            SQLDataType::Timestamp(resolution, tz) => {
+                match tz {
+                    TimezoneInfo::None => {},
+                    _ => {
+                        return Err(Error::IO {
+                            message: format!("Timezone not supported in timestamp"),
+                        })
+                    }
+                };
+                let time_unit = match resolution {
+                    // Default to microsecond to match PyArrow
+                    None => TimeUnit::Microsecond,
+                    Some(0) => TimeUnit::Second,
+                    Some(3) => TimeUnit::Millisecond,
+                    Some(6) => TimeUnit::Microsecond,
+                    Some(9) => TimeUnit::Nanosecond,
+                    _ => {
+                        return Err(Error::IO {
+                            message: format!("Unsupported datetime resolution: {:?}", resolution),
+                        })
+                    }
+                };
+                Ok(ArrowDataType::Timestamp(time_unit, None))
+            }
+            SQLDataType::Datetime(resolution) => {
+                let time_unit = match resolution {
+                    None => TimeUnit::Microsecond,
+                    Some(0) => TimeUnit::Second,
+                    Some(3) => TimeUnit::Millisecond,
+                    Some(6) => TimeUnit::Microsecond,
+                    Some(9) => TimeUnit::Nanosecond,
+                    _ => {
+                        return Err(Error::IO {
+                            message: format!("Unsupported datetime resolution: {:?}", resolution),
+                        })
+                    }
+                };
+                Ok(ArrowDataType::Timestamp(time_unit, None))
+            }
+            _ => Err(Error::IO {
+                message: format!("Unsupported data type: {:?}. Supports: float, double, boolean, int, integer, date, datetime", data_type),
```

Comment:
Also supports binary / string / booleans, right? I mean just in the error message.

Reply:
Updated